app/login.go

```diff
@@ -245,30 +246,37 @@ func (a *App) AttachCloudSessionCookie(c *request.Context, w http.ResponseWriter
 	}
 
 	maxAge := *a.Config().ServiceSettings.SessionLengthWebInDays * 60 * 60 * 24
-	domain := a.GetCookieDomain()
 	subpath, _ := utils.GetSubpathFromConfig(a.Config())
 	expiresAt := time.Unix(model.GetMillis()/1000+int64(maxAge), 0)
 
-	var val string
-	if strings.Contains(domain, "localhost") {
-		val = "localhost"
-	} else {
-		val = strings.SplitN(domain, ".", 2)[0]
-		domain = strings.SplitN(domain, ".", 3)[2]
+	domain := ""
+	if siteURL, err := url.Parse(a.GetSiteURL()); err == nil {
+		domain = siteURL.Hostname()
 	}
 
-	cookie := &http.Cookie{
-		Name:     model.SessionCookieCloudUrl,
-		Value:    val,
-		Path:     subpath,
-		MaxAge:   maxAge,
-		Expires:  expiresAt,
-		HttpOnly: true,
-		Domain:   domain,
-		Secure:   secure,
-	}
+	if domain != "" {
+		var val string
+		if strings.Contains(domain, "localhost") {
+			val = "localhost"
+		} else {
+			val = strings.SplitN(domain, ".", 2)[0]
+			domain = strings.SplitN(domain, ".", 3)[2]
+			domain = "." + domain
```

Comment:
If you pass `test.com` as the domain value, this will panic. Maybe we should wrap this in a cloud check?

Reply:
There's a cloud check in the api layer to ensure this function only gets called for cloud. For this, all our cloud workspace URLs in the form of `example.cloud.mattermost.com`.

Reply:
Here's a little test for panic: https://goplay.tools/snippet/_abLj1J_VNj

Reply:
Yes, it's true but our cloud URLs can never be in the form `example.com` or `test.com`